### PR TITLE
Enhancement: Add enablePools option to TrueNAS service widget

### PR DESCRIPTION
**Pull request by dstotz**
## Proposed change

- Maintain the previous behavior with previous configuration
- Add new optional parameter for `enablePools` that when enabled will display each Pool in the TrueNAS instance with its disk usage, and a status indicator

image

Closes #2792 

## Type of change

- [ ] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [X] If applicable, I have added corresponding documentation changes.
- [X] If applicable, I have reviewed the [feature](https://gethomepage.dev/latest/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [X] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/latest/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/latest/more/development/#code-linting).
- [X] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

File: src/widgets/truenas/proxy.js
Comment (shamoon) [inline src/widgets/truenas/proxy.js:1]: This proxy seems unnecessary, I need to check but I think you can just use credentialed

File: src/utils/proxy/handlers/credentialed.js [hunk 1]
```diff
@@ -70,6 +70,7 @@ export default async function credentialedProxyHandler(req, res, map) {
         withCredentials: true,
         credentials: "include",
         headers,
+        body: req.body,
```
Comment (shamoon) [inline src/utils/proxy/handlers/credentialed.js:73]: Again, needed?

**Review by shamoon — CHANGES_REQUESTED** [review]
Also would be great if you can supply example api output

**Comment by dstotz** [comment]
> Also would be great if you can supply example api output

Sure, no problem! Do you wan the example as an inline comment somewhere in the code? Or just here on the PR for your reference? Here is a link to the API docs https://www.truenas.com/docs/api/scale_rest_api.html it is the `GET /pool` request

File: src/widgets/truenas/proxy.js
Comment (dstotz) [reply src/widgets/truenas/proxy.js:1]: Agreed. I had just moved the previous request logic out of the existing `widget.js` to clean stuff up, but you are right we just don't need it at all.

Done.

**Comment by shamoon** [comment]
> Sure, no problem! Do you wan the example as an inline comment somewhere in the code? Or just here on the PR for your reference? Here is a link to the API docs https://www.truenas.com/docs/api/scale_rest_api.html it is the `GET /pool` request

Just in a comment here for testing

image

**Comment by shamoon** [comment]
You re-requested review but didnt post API output, unless Im not seeing it?

**Comment by dstotz** [comment]
> You re-requested review but didnt post API output, unless Im not seeing it?

Apologies, miscommunication. I had thought your screenshot indicated you found what you need by looking at the response schema. I'll include an output example below

```
[
    {
        "id": 3,
        "name": "AppData",
        "guid": "12345fakeguid67890",
        "encrypt": 0,
        "encryptkey": "",
        "path": "/mnt/AppData",
        "status": "ONLINE",
        "scan": {
            "function": "SCRUB",
            "state": "FINISHED",
            "start_time": {
                "$date": 1705816803000
            },
            "end_time": {
                "$date": 1705817857000
            },
            "percentage": 99.99964237213135,
            "bytes_to_process": 441097035776,
            "bytes_processed": 441098600448,
            "bytes_issued": 441097035776,
            "pause": null,
            "errors": 0,
            "total_secs_left": null
        },
        "topology": {
            "data": [
                {
                    "name": "mirror-0",
                    "type": "MIRROR",
                    "path": null,
                    "guid": "12345fakeguid67890",
                    "status": "ONLINE",
                    "stats": {
                        "timestamp": 129768022889925,
                        "read_errors": 0,
                        "write_errors": 0,
                        "checksum_errors": 0,
                        "ops": [
                            0,
                            478115,
                            7398499,
                            0,
                            0,
                            0,
                            0
                        ],
                        "bytes": [
                            0,
                            5816020992,
                            194377064448,
                            0,
                            0,
                            0,
                            0
                        ],
                        "size": 996432412672,
                        "allocated": 503696805888,
                        "fragmentation": 47,
                        "self_healed": 0,
                        "configured_ashift": 12,
                        "logical_ashift": 9,
                        "physical_ashift": 9
                    },
                    "children": [
                        {
                            "name": "fake-uuid-value-dummy-data",
                            "type": "DISK",
                            "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                            "guid": "12345fakeguid67890",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 129768022939568,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    236984,
                                    3688201,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    2854191104,
                                    97188532224,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 9
                            },
                            "children": [],
                            "device": "sdi2",
                            "disk": "sdi",
                            "unavail_disk": null
                        },
                        {
                            "name": "fake-uuid-value-dummy-data",
                            "type": "DISK",
                            "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                            "guid": "12345fakeguid67890",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 129768022968122,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    241131,
                                    3710298,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    2961829888,
                                    97188532224,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 9
                            },
                            "children": [],
                            "device": "sdh2",
                            "disk": "sdh",
                            "unavail_disk": null
                        }
                    ],
                    "unavail_disk": null
                }
            ],
            "log": [],
            "cache": [],
            "spare": [],
            "special": [],
            "dedup": []
        },
        "healthy": true,
        "warning": false,
        "status_detail": null,
        "size": 996432412672,
        "allocated": 503696805888,
        "free": 492735606784,
        "freeing": 0,
        "fragmentation": "47",
        "size_str": "996432412672",
        "allocated_str": "503696805888",
        "free_str": "492735606784",
        "freeing_str": "0",
        "autotrim": {
            "value": "off",
            "rawvalue": "off",
            "parsed": "off",
            "source": "DEFAULT"
        },
        "encryptkey_path": null,
        "is_decrypted": true
    },
    {
        "id": 4,
        "name": "Fun",
        "guid": "12345fakeguid67890",
        "encrypt": 0,
        "encryptkey": "",
        "path": "/mnt/Fun",
        "status": "ONLINE",
        "scan": {
            "function": "SCRUB",
            "state": "FINISHED",
            "start_time": {
                "$date": 1705816812000
            },
            "end_time": {
                "$date": 1705816921000
            },
            "percentage": 99.54779148101807,
            "bytes_to_process": 2342563840,
            "bytes_processed": 2353205248,
            "bytes_issued": 2342563840,
            "pause": null,
            "errors": 0,
            "total_secs_left": null
        },
        "topology": {
            "data": [
                {
                    "name": "fake-uuid-value-dummy-data",
                    "type": "DISK",
                    "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                    "guid": "12345fakeguid67890",
                    "status": "ONLINE",
                    "stats": {
                        "timestamp": 129766329160904,
                        "read_errors": 0,
                        "write_errors": 0,
                        "checksum_errors": 0,
                        "ops": [
                            0,
                            12695,
                            1431842,
                            0,
                            0,
                            0,
                            0
                        ],
                        "bytes": [
                            0,
                            1773674496,
                            41797828608,
                            0,
                            0,
                            0,
                            0
                        ],
                        "size": 1992864825344,
                        "allocated": 2241433600,
                        "fragmentation": 0,
                        "self_healed": 0,
                        "configured_ashift": 12,
                        "logical_ashift": 9,
                        "physical_ashift": 12
                    },
                    "children": [],
                    "device": "sde2",
                    "disk": "sde",
                    "unavail_disk": null
                }
            ],
            "log": [],
            "cache": [],
            "spare": [],
            "special": [],
            "dedup": []
        },
        "healthy": true,
        "warning": false,
        "status_detail": null,
        "size": 1992864825344,
        "allocated": 2241433600,
        "free": 1990623391744,
        "freeing": 0,
        "fragmentation": "0",
        "size_str": "1992864825344",
        "allocated_str": "2241433600",
        "free_str": "1990623391744",
        "freeing_str": "0",
        "autotrim": {
            "value": "off",
            "rawvalue": "off",
            "parsed": "off",
            "source": "DEFAULT"
        },
        "encryptkey_path": null,
        "is_decrypted": true
    },
    {
        "id": 5,
        "name": "Tank",
        "guid": "12345fakeguid67890",
        "encrypt": 0,
        "encryptkey": "",
        "path": "/mnt/Tank",
        "status": "ONLINE",
        "scan": {
            "function": "SCRUB",
            "state": "FINISHED",
            "start_time": {
                "$date": 1707026401000
            },
            "end_time": {
                "$date": 1707048033000
            },
            "percentage": 99.99997019767761,
            "bytes_to_process": 20692078739456,
            "bytes_processed": 20692083965952,
            "bytes_issued": 20692078739456,
            "pause": null,
            "errors": 0,
            "total_secs_left": null
        },
        "topology": {
            "data": [
                {
                    "name": "raidz1-0",
                    "type": "RAIDZ1",
                    "path": null,
                    "guid": "12345fakeguid67890",
                    "status": "ONLINE",
                    "stats": {
                        "timestamp": 129753743825836,
                        "read_errors": 0,
                        "write_errors": 0,
                        "checksum_errors": 0,
                        "ops": [
                            0,
                            902085,
                            1690960,
                            0,
                            0,
                            0,
                            0
                        ],
                        "bytes": [
                            0,
                            32848642048,
                            28755333120,
                            0,
                            0,
                            0,
                            0
                        ],
                        "size": 39994735460352,
                        "allocated": 20770749317120,
                        "fragmentation": 0,
                        "self_healed": 0,
                        "configured_ashift": 12,
                        "logical_ashift": 9,
                        "physical_ashift": 9
                    },
                    "children": [
                        {
                            "name": "fake-uuid-value-dummy-data",
                            "type": "DISK",
                            "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                            "guid": "12345fakeguid67890",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 129753743878495,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    176209,
                                    338077,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    6440026112,
                                    5750685696,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 9
                            },
                            "children": [],
                            "device": "sdc2",
                            "disk": "sdc",
                            "unavail_disk": null
                        },
                        {
                            "name": "fake-uuid-value-dummy-data",
                            "type": "DISK",
                            "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                            "guid": "12345fakeguid67890",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 129753743911406,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    180334,
                                    337910,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    6548520960,
                                    5751091200,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 9
                            },
                            "children": [],
                            "device": "sdf2",
                            "disk": "sdf",
                            "unavail_disk": null
                        },
                        {
                            "name": "fake-uuid-value-dummy-data",
                            "type": "DISK",
                            "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                            "guid": "12345fakeguid67890",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 129753743944799,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    180022,
                                    338298,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    6558081024,
                                    5751521280,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 9
                            },
                            "children": [],
                            "device": "sdd2",
                            "disk": "sdd",
                            "unavail_disk": null
                        },
                        {
                            "name": "fake-uuid-value-dummy-data",
                            "type": "DISK",
                            "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                            "guid": "12345fakeguid67890",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 129753743982109,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    181803,
                                    338725,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    6597038080,
                                    5752324096,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 9
                            },
                            "children": [],
                            "device": "sdb2",
                            "disk": "sdb",
                            "unavail_disk": null
                        },
                        {
                            "name": "fake-uuid-value-dummy-data",
                            "type": "DISK",
                            "path": "/dev/disk/by-partuuid/fake-uuid-value-dummy-data",
                            "guid": "12345fakeguid67890",
                            "status": "ONLINE",
                            "stats": {
                                "timestamp": 129753744018417,
                                "read_errors": 0,
                                "write_errors": 0,
                                "checksum_errors": 0,
                                "ops": [
                                    0,
                                    183717,
                                    337950,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "bytes": [
                                    0,
                                    6704975872,
                                    5749710848,
                                    0,
                                    0,
                                    0,
                                    0
                                ],
                                "size": 0,
                                "allocated": 0,
                                "fragmentation": 0,
                                "self_healed": 0,
                                "configured_ashift": 12,
                                "logical_ashift": 9,
                                "physical_ashift": 9
                            },
                            "children": [],
                            "device": "sda2",
                            "disk": "sda",
                            "unavail_disk": null
                        }
                    ],
                    "unavail_disk": null
                }
            ],
            "log": [],
            "cache": [],
            "spare": [],
            "special": [],
            "dedup": []
        },
        "healthy": true,
        "warning": false,
        "status_detail": null,
        "size": 39994735460352,
        "allocated": 20770749317120,
        "free": 19223986143232,
        "freeing": 0,
        "fragmentation": "0",
        "size_str": "39994735460352",
        "allocated_str": "20770749317120",
        "free_str": "19223986143232",
        "freeing_str": "0",
        "autotrim": {
            "value": "off",
            "rawvalue": "off",
            "parsed": "off",
            "source": "DEFAULT"
        },
        "encryptkey_path": null,
        "is_decrypted": true
    }
]
```

**Review by shamoon — APPROVED** [review]
Thanks. Yea I think another remnant of your other PR, I moved the pool component. Also it really bugged me that the % didnt line up on the far right, sorry to be crazy.

But works great, thanks

Screenshot 2024-02-15 at 9 29 37 PM

**Comment by fayimora** [comment]
I have the same issue as @gco .
```json
[{"id":"tank","name":"tank","healthy":true}]
```

**Comment by gco** [comment]
@fayimora opened a discussion in #3129

**Pull request merged**